test/e2e/upgrade_management_cluster_test.go

```diff
@@ -0,0 +1,352 @@
+	output, err = cmd.CombinedOutput()
+	Logf("scale: %v", string(output))
+	Expect(err).To(BeNil())
+
+	Logf("Waiting for 2 BMHs to be in Available state")
+	Eventually(func(g Gomega) {
+		bmhs, err := getAllBmhs(ctx, upgradeClusterClient, namespace, specName)
+		g.Expect(err).To(BeNil())
+		g.Expect(filterBmhsByProvisioningState(bmhs, bmo.StateAvailable)).To(HaveLen(2))
+	}, e2eConfig.GetIntervals(specName, "wait-machine-remediation")...).Should(Succeed())
+
+	/*-------------------------------*
+	| Create a test workload cluster |
+	*--------------------------------*/
+	var (
+		controlPlaneMachineCount = pointer.Int64Ptr(1)
+		workerMachineCount       = pointer.Int64Ptr(1)
+	)
+	By("Creating a test workload cluster")
+
+	Logf("Creating the workload cluster with name %q using the %q template (Kubernetes %s, %d control-plane machines, %d worker machines)",
+		workloadClusterName, "(default)", kubernetesVersion, *controlPlaneMachineCount, *workerMachineCount)
+
+	Logf("Getting the cluster template yaml")
+	Logf("clusterctlBinaryPath = %v", clusterctlBinaryPath)
+	Logf("ClusterctlConfigPath = %v", clusterctlConfigPath)
+	workloadClusterTemplate := clusterctl.ConfigClusterWithBinary(ctx, clusterctlBinaryPath, clusterctl.ConfigClusterInput{
+		// pass reference to the management cluster hosting this test
+		KubeconfigPath: upgradeClusterProxy.GetKubeconfigPath(),
+		// pass the clusterctl config file that points to the local provider repository created for this test,
+
+		ClusterctlConfigPath: clusterctlConfigPath,
+
+		// define template variables
+		Namespace:                namespace,
+		ClusterName:              workloadClusterName,
+		Flavor:                   "ha-upgrade-workload",
+		KubernetesVersion:        kubernetesVersion,
+		ControlPlaneMachineCount: controlPlaneMachineCount,
+		WorkerMachineCount:       workerMachineCount,
+		InfrastructureProvider:   clusterctl.DefaultInfrastructureProvider,
+		// setup clusterctl logs folder
+		LogFolder: filepath.Join(artifactFolder, "clusters", upgradeClusterProxy.GetName()),
+	})
+	Expect(workloadClusterTemplate).ToNot(BeNil(), "Failed to get the cluster template")
+	Logf("%v ", string(workloadClusterTemplate))
+
+	Logf("Applying the cluster template yaml to the cluster")
+	Expect(upgradeClusterProxy.Apply(ctx, workloadClusterTemplate)).To(Succeed())
+
+	By("Waiting for the machines to be running")
+	Eventually(func() (int64, error) {
+		var n int64
+		machineList := &clusterv1old.MachineList{}
+		if err := upgradeClusterClient.List(ctx, machineList, client.InNamespace(namespace), client.MatchingLabels{clusterv1.ClusterLabelName: workloadClusterName}); err == nil {
+			for _, machine := range machineList.Items {
+				if strings.EqualFold(machine.Status.Phase, "running") {
+					n++
+				}
+			}
+		}
+		return n, nil
+	}, e2eConfig.GetIntervals(specName, "wait-worker-nodes")...).Should(Equal(*controlPlaneMachineCount))
+
+	By("THE MANAGEMENT CLUSTER WITH OLDER VERSION OF PROVIDERS WORKS!")
```

Comment:
After sleeping on this I think we should remove this part. We don't do this in the ansible upgrade test and I would prefer to keep it the same here to start with. We also already test that the old version can create a cluster in main integration tests.

I think there was a bit of confusion about this maybe because I didn't explain it clearly before. Earlier we didn't do pivoting in this PR and we also didn't have the "second workload cluster" so there was not really any test to show that the upgraded components actually worked. Now that we have pivoting in here that will be enough to make sure that things are working. So great job adding that pivot! :tada:

What I'm trying to say is basically that we don't need both pivot and a second cluster, it is enough to "just" have pivot.